[COMPILE-ERROR] MapReduce tests with jdk-6

```
15:08:52 [ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:2.5.1:testCompile (default-testCompile) on project hazelcast: Compilation failure: Compilation failure:
15:08:52 [ERROR] /scratch/jenkins/workspace/Hazelcast-3.x-OracleJDK1.6/hazelcast/src/test/java/com/hazelcast/mapreduce/MapReduceTest.java:[208,54] incompatible types
15:08:52 [ERROR] found   : com.hazelcast.mapreduce.Job<java.lang.Object,java.lang.Integer>
15:08:52 [ERROR] required: com.hazelcast.mapreduce.Job<java.lang.Integer,java.lang.Integer>
15:08:52 [ERROR] /scratch/jenkins/workspace/Hazelcast-3.x-OracleJDK1.6/hazelcast/src/test/java/com/hazelcast/mapreduce/MapReduceTest.java:[251,54] incompatible types
15:08:52 [ERROR] found   : com.hazelcast.mapreduce.Job<java.lang.Object,java.lang.Integer>
15:08:52 [ERROR] required: com.hazelcast.mapreduce.Job<java.lang.Integer,java.lang.Integer>
15:08:52 [ERROR] /scratch/jenkins/workspace/Hazelcast-3.x-OracleJDK1.6/hazelcast/src/test/java/com/hazelcast/mapreduce/MapReduceTest.java:[286,54] incompatible types
15:08:52 [ERROR] found   : com.hazelcast.mapreduce.Job<java.lang.Object,java.lang.Integer>
15:08:52 [ERROR] required: com.hazelcast.mapreduce.Job<java.lang.Integer,java.lang.Integer>
15:08:52 [ERROR] /scratch/jenkins/workspace/Hazelcast-3.x-OracleJDK1.6/hazelcast/src/test/java/com/hazelcast/mapreduce/MapReduceTest.java:[322,54] incompatible types
15:08:52 [ERROR] found   : com.hazelcast.mapreduce.Job<java.lang.Object,java.lang.Integer>
15:08:52 [ERROR] required: com.hazelcast.mapreduce.Job<java.lang.Integer,java.lang.Integer>
15:08:52 [ERROR] /scratch/jenkins/workspace/Hazelcast-3.x-OracleJDK1.6/hazelcast/src/test/java/com/hazelcast/mapreduce/MapReduceTest.java:[355,54] incompatible types
15:08:52 [ERROR] found   : com.hazelcast.mapreduce.Job<java.lang.Object,java.lang.Integer>
15:08:52 [ERROR] required: com.hazelcast.mapreduce.Job<java.lang.Integer,java.lang.Integer>
15:08:52 [ERROR] /scratch/jenkins/workspace/Hazelcast-3.x-OracleJDK1.6/hazelcast/src/test/java/com/hazelcast/mapreduce/MapReduceTest.java:[385,54] incompatible types
15:08:52 [ERROR] found   : com.hazelcast.mapreduce.Job<java.lang.Object,java.lang.Integer>
15:08:52 [ERROR] required: com.hazelcast.mapreduce.Job<java.lang.Integer,java.lang.Integer>
15:08:52 [ERROR] /scratch/jenkins/workspace/Hazelcast-3.x-OracleJDK1.6/hazelcast/src/test/java/com/hazelcast/mapreduce/MapReduceTest.java:[418,54] incompatible types
15:08:52 [ERROR] found   : com.hazelcast.mapreduce.Job<java.lang.Object,java.lang.Integer>
15:08:52 [ERROR] required: com.hazelcast.mapreduce.Job<java.lang.Integer,java.lang.Integer>
15:08:52 [ERROR] /scratch/jenkins/workspace/Hazelcast-3.x-OracleJDK1.6/hazelcast/src/test/java/com/hazelcast/mapreduce/MapReduceTest.java:[449,54] incompatible types
15:08:52 [ERROR] found   : com.hazelcast.mapreduce.Job<java.lang.Object,java.lang.Integer>
15:08:52 [ERROR] required: com.hazelcast.mapreduce.Job<java.lang.Integer,java.lang.Integer>
.....
```
https://hazelcast-l337.ci.cloudbees.com/view/Official%20Builds/job/Hazelcast-3.x-OracleJDK1.6/721/console

--- a/hazelcast/src/test/java/com/hazelcast/mapreduce/DistributedMapperMapReduceTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/mapreduce/DistributedMapperMapReduceTest.java
@@ -31,6 +31,7 @@ import java.util.HashMap;
 import java.util.Map;
 import java.util.concurrent.Semaphore;
 
+import static com.hazelcast.mapreduce.MapReduceTest.integerKvSource;
 import static org.junit.Assert.assertEquals;
 
 @RunWith(HazelcastParallelClassRunner.class)
@@ -58,7 +59,7 @@ public class DistributedMapperMapReduceTest
         }
 
         JobTracker tracker = h1.getJobTracker("default");
-        Job<Integer, Integer> job = tracker.newJob(KeyValueSource.fromMap(m1));
+        Job<Integer, Integer> job = tracker.newJob(integerKvSource(m1));
         ICompletableFuture<Map<String, Integer>> future = job.mapper(new GroupingTestMapper()).combiner(new TestCombinerFactory())
                                                              .reducer(new TestReducerFactory()).submit();
 
@@ -93,7 +94,7 @@ public class DistributedMapperMapReduceTest
         }
 
         JobTracker tracker = h1.getJobTracker("default");
-        Job<Integer, Integer> job = tracker.newJob(KeyValueSource.fromMap(m1));
+        Job<Integer, Integer> job = tracker.newJob(integerKvSource(m1));
         ICompletableFuture<Map<String, Long>> future = job.mapper(new GroupingTestMapper())
                                                           .combiner(new TestIntermediateCombinerFactory())
                                                           .reducer(new TestIntermediateReducerFactory()).submit();
@@ -129,7 +130,7 @@ public class DistributedMapperMapReduceTest
         }
 
         JobTracker tracker = h1.getJobTracker("default");
-        Job<Integer, Integer> job = tracker.newJob(KeyValueSource.fromMap(m1));
+        Job<Integer, Integer> job = tracker.newJob(integerKvSource(m1));
         ICompletableFuture<Integer> future = job.mapper(new GroupingTestMapper()).combiner(new TestCombinerFactory())
                                                 .reducer(new TestReducerFactory()).submit(new TestCollator());
 
@@ -167,7 +168,7 @@ public class DistributedMapperMapReduceTest
         semaphore.acquire();
 
         JobTracker tracker = h1.getJobTracker("default");
-        Job<Integer, Integer> job = tracker.newJob(KeyValueSource.fromMap(m1));
+        Job<Integer, Integer> job = tracker.newJob(integerKvSource(m1));
         ICompletableFuture<Map<String, Integer>> future = job.mapper(new GroupingTestMapper()).combiner(new TestCombinerFactory())
                                                              .reducer(new TestReducerFactory()).submit();
 
@@ -219,7 +220,7 @@ public class DistributedMapperMapReduceTest
         semaphore.acquire();
 
         JobTracker tracker = h1.getJobTracker("default");
-        Job<Integer, Integer> job = tracker.newJob(KeyValueSource.fromMap(m1));
+        Job<Integer, Integer> job = tracker.newJob(integerKvSource(m1));
         ICompletableFuture<Integer> future = job.mapper(new GroupingTestMapper()).combiner(new TestCombinerFactory())
                                                 .reducer(new TestReducerFactory()).submit(new TestCollator());
 

--- a/hazelcast/src/test/java/com/hazelcast/mapreduce/MapReduceLiteMemberTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/mapreduce/MapReduceLiteMemberTest.java
@@ -19,6 +19,7 @@ import org.junit.runner.RunWith;
 import java.util.List;
 import java.util.Map;
 
+import static com.hazelcast.mapreduce.MapReduceTest.integerKvSource;
 import static org.junit.Assert.assertEquals;
 
 @RunWith(HazelcastParallelClassRunner.class)
@@ -246,7 +247,7 @@ public class MapReduceLiteMemberTest
         final String mapName = randomMapName();
         JobTracker tracker = instance.getJobTracker(mapName);
         final IMap<Integer, Integer> m1 = instance.getMap(mapName);
-        Job<Integer, Integer> job = tracker.newJob(KeyValueSource.fromMap(m1));
+        Job<Integer, Integer> job = tracker.newJob(integerKvSource(m1));
         return job.mapper(new MapReduceTest.TestMapper()).submit();
     }
 
@@ -258,7 +259,7 @@ public class MapReduceLiteMemberTest
         }
 
         JobTracker tracker = instance.getJobTracker(mapName);
-        return tracker.newJob(KeyValueSource.fromMap(m1));
+        return tracker.newJob(integerKvSource(m1));
     }
 
 }

--- a/hazelcast/src/test/java/com/hazelcast/mapreduce/MapReduceManagedContextTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/mapreduce/MapReduceManagedContextTest.java
@@ -65,7 +65,7 @@ public class MapReduceManagedContextTest
         }
 
         JobTracker tracker = h1.getJobTracker("default");
-        KeyValueSource<String, Integer> source = KeyValueSource.fromMap(m1);
+        KeyValueSource<String, Integer> source = KeyValueSource.<String, Integer>fromMap(m1);
         KeyValueSource<String, Integer> wrapper = new InjectedKeyValueSource<String, Integer>(source);
         Job<String, Integer> job = tracker.newJob(wrapper);
 

--- a/hazelcast/src/test/java/com/hazelcast/mapreduce/MapReduceStressTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/mapreduce/MapReduceStressTest.java
@@ -36,6 +36,7 @@ import java.util.Random;
 import java.util.concurrent.CountDownLatch;
 import java.util.concurrent.atomic.AtomicReferenceArray;
 
+import static com.hazelcast.mapreduce.MapReduceTest.integerKvSource;
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.fail;
 
@@ -125,7 +126,7 @@ public class MapReduceStressTest
         @Override
         public void run() {
             JobTracker jobTracker = hazelcastInstance.getJobTracker(HazelcastTestSupport.randomString());
-            Job<Integer, Integer> job = jobTracker.newJob(KeyValueSource.fromMap(map));
+            Job<Integer, Integer> job = jobTracker.newJob(integerKvSource(map));
             JobCompletableFuture<Map<String, BigInteger>> future = job.chunkSize(0).mapper(new GroupingTestMapper())
                                                                       .combiner(new ObjectCombinerFactory())
                                                                       .reducer(new ObjectReducerFactory()).submit();

--- a/hazelcast/src/test/java/com/hazelcast/mapreduce/MapReduceTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/mapreduce/MapReduceTest.java
@@ -91,7 +91,7 @@ public class MapReduceTest extends HazelcastTestSupport {
             }
 
             JobTracker tracker = h1.getJobTracker("default");
-            KeyValueSource<Integer, Integer> kvs = KeyValueSource.fromMap(m1);
+            KeyValueSource<Integer, Integer> kvs = integerKvSource(m1);
             KeyValueSource<Integer, Integer> wrapper = new MapKeyValueSourceAdapter<Integer, Integer>(kvs);
             Job<Integer, Integer> job = tracker.newJob(wrapper);
 
@@ -131,7 +131,7 @@ public class MapReduceTest extends HazelcastTestSupport {
             }
 
             JobTracker tracker = h1.getJobTracker("default");
-            KeyValueSource<Integer, Integer> kvs = KeyValueSource.fromMap(m1);
+            KeyValueSource<Integer, Integer> kvs = integerKvSource(m1);
             KeyValueSource<Integer, Integer> wrapper = new MapKeyValueSourceAdapter<Integer, Integer>(kvs);
             Job<Integer, Integer> job = tracker.newJob(wrapper);
             ICompletableFuture<Map<String, List<Integer>>> future = job.mapper(new TestMapper()).submit();
@@ -166,7 +166,7 @@ public class MapReduceTest extends HazelcastTestSupport {
             }
 
             JobTracker tracker = h1.getJobTracker("default");
-            KeyValueSource<Integer, Integer> kvs = KeyValueSource.fromMap(m1);
+            KeyValueSource<Integer, Integer> kvs = integerKvSource(m1);
             KeyValueSource<Integer, Integer> wrapper = new MapKeyValueSourceAdapter<Integer, Integer>(kvs);
             Job<Integer, Integer> job = tracker.newJob(wrapper);
             ICompletableFuture<Map<String, List<Integer>>> future =
@@ -205,7 +205,7 @@ public class MapReduceTest extends HazelcastTestSupport {
             }
 
             JobTracker tracker = h1.getJobTracker("default");
-            Job<Integer, Integer> job = tracker.newJob(KeyValueSource.fromMap(m1));
+            Job<Integer, Integer> job = tracker.newJob(integerKvSource(m1));
             ICompletableFuture<Map<String, List<Integer>>> future = job
                     .mapper(new ExceptionThrowingMapper())
                     .submit(new Collator<Map.Entry<String, List<Integer>>, Map<String, List<Integer>>>() {
@@ -248,7 +248,7 @@ public class MapReduceTest extends HazelcastTestSupport {
             }
 
             JobTracker tracker = h1.getJobTracker("default");
-            Job<Integer, Integer> job = tracker.newJob(KeyValueSource.fromMap(m1));
+            Job<Integer, Integer> job = tracker.newJob(integerKvSource(m1));
             ICompletableFuture<Map<String, List<Integer>>> future = job.mapper(new ExceptionThrowingMapper()).submit();
 
             try {
@@ -283,7 +283,7 @@ public class MapReduceTest extends HazelcastTestSupport {
             }
 
             JobTracker tracker = h1.getJobTracker("default");
-            Job<Integer, Integer> job = tracker.newJob(KeyValueSource.fromMap(m1));
+            Job<Integer, Integer> job = tracker.newJob(integerKvSource(m1));
             ICompletableFuture<Map<String, List<Integer>>> future = job.mapper(new TimeConsumingMapper()).submit();
 
             future.cancel(true);
@@ -319,7 +319,7 @@ public class MapReduceTest extends HazelcastTestSupport {
             }
 
             JobTracker tracker = h1.getJobTracker("default");
-            Job<Integer, Integer> job = tracker.newJob(KeyValueSource.fromMap(m1));
+            Job<Integer, Integer> job = tracker.newJob(integerKvSource(m1));
             ICompletableFuture<Map<String, List<Integer>>> future = job.mapper(new TestMapper()).submit();
 
             Map<String, List<Integer>> result = future.get();
@@ -352,7 +352,7 @@ public class MapReduceTest extends HazelcastTestSupport {
             }
 
             JobTracker tracker = h1.getJobTracker("default");
-            Job<Integer, Integer> job = tracker.newJob(KeyValueSource.fromMap(m1));
+            Job<Integer, Integer> job = tracker.newJob(integerKvSource(m1));
             ICompletableFuture<Integer> future = job.onKeys(50).mapper(new TestMapper()).submit(new GroupingTestCollator());
 
             int result = future.get();
@@ -382,7 +382,7 @@ public class MapReduceTest extends HazelcastTestSupport {
             }
 
             JobTracker tracker = h1.getJobTracker("default");
-            Job<Integer, Integer> job = tracker.newJob(KeyValueSource.fromMap(m1));
+            Job<Integer, Integer> job = tracker.newJob(integerKvSource(m1));
             ICompletableFuture<Integer> future = job
                     .keyPredicate(new TestKeyPredicate())
                     .mapper(new TestMapper())
@@ -415,7 +415,7 @@ public class MapReduceTest extends HazelcastTestSupport {
             }
 
             JobTracker tracker = h1.getJobTracker("default");
-            Job<Integer, Integer> job = tracker.newJob(KeyValueSource.fromMap(m1));
+            Job<Integer, Integer> job = tracker.newJob(integerKvSource(m1));
             ICompletableFuture<Map<String, List<Integer>>> future = job.mapper(new GroupingTestMapper(2)).submit();
 
             Map<String, List<Integer>> result = future.get();
@@ -446,7 +446,7 @@ public class MapReduceTest extends HazelcastTestSupport {
             }
 
             JobTracker tracker = h1.getJobTracker("default");
-            Job<Integer, Integer> job = tracker.newJob(KeyValueSource.fromMap(m1));
+            Job<Integer, Integer> job = tracker.newJob(integerKvSource(m1));
             ICompletableFuture<Map<String, Integer>> future = job
                     .mapper(new GroupingTestMapper())
                     .reducer(new TestReducerFactory())
@@ -488,7 +488,7 @@ public class MapReduceTest extends HazelcastTestSupport {
             }
 
             JobTracker tracker = h1.getJobTracker("default");
-            Job<Integer, Integer> job = tracker.newJob(KeyValueSource.fromMap(m1));
+            Job<Integer, Integer> job = tracker.newJob(integerKvSource(m1));
             JobCompletableFuture<Map<String, Integer>> future = job
                     .chunkSize(10)
                     .mapper(new GroupingTestMapper())
@@ -543,7 +543,7 @@ public class MapReduceTest extends HazelcastTestSupport {
             }
 
             JobTracker tracker = h1.getJobTracker("default");
-            Job<Integer, Integer> job = tracker.newJob(KeyValueSource.fromMap(m1));
+            Job<Integer, Integer> job = tracker.newJob(integerKvSource(m1));
             ICompletableFuture<Integer> future = job.mapper(new GroupingTestMapper()).submit(new GroupingTestCollator());
 
             int result = future.get();
@@ -581,7 +581,7 @@ public class MapReduceTest extends HazelcastTestSupport {
             }
 
             JobTracker tracker = h1.getJobTracker("default");
-            Job<Integer, Integer> job = tracker.newJob(KeyValueSource.fromMap(m1));
+            Job<Integer, Integer> job = tracker.newJob(integerKvSource(m1));
             ICompletableFuture<Integer> future = job.mapper(new GroupingTestMapper()).reducer(new TestReducerFactory())
                     .submit(new TestCollator());
 
@@ -624,7 +624,7 @@ public class MapReduceTest extends HazelcastTestSupport {
             semaphore.acquire();
 
             JobTracker tracker = h1.getJobTracker("default");
-            Job<Integer, Integer> job = tracker.newJob(KeyValueSource.fromMap(m1));
+            Job<Integer, Integer> job = tracker.newJob(integerKvSource(m1));
             ICompletableFuture<Map<String, List<Integer>>> future = job.mapper(new TestMapper()).submit();
 
             future.andThen(new ExecutionCallback<Map<String, List<Integer>>>() {
@@ -677,7 +677,7 @@ public class MapReduceTest extends HazelcastTestSupport {
             semaphore.acquire();
 
             JobTracker tracker = h1.getJobTracker("default");
-            Job<Integer, Integer> job = tracker.newJob(KeyValueSource.fromMap(m1));
+            Job<Integer, Integer> job = tracker.newJob(integerKvSource(m1));
             ICompletableFuture<Map<String, List<Integer>>> future = job.onKeys(50).mapper(new TestMapper()).submit();
 
             future.andThen(new ExecutionCallback<Map<String, List<Integer>>>() {
@@ -731,7 +731,7 @@ public class MapReduceTest extends HazelcastTestSupport {
             semaphore.acquire();
 
             JobTracker tracker = h1.getJobTracker("default");
-            Job<Integer, Integer> job = tracker.newJob(KeyValueSource.fromMap(m1));
+            Job<Integer, Integer> job = tracker.newJob(integerKvSource(m1));
             ICompletableFuture<Map<String, Integer>> future = job.mapper(new GroupingTestMapper())
                     .reducer(new TestReducerFactory()).submit();
 
@@ -791,7 +791,7 @@ public class MapReduceTest extends HazelcastTestSupport {
             semaphore.acquire();
 
             JobTracker tracker = h1.getJobTracker("default");
-            Job<Integer, Integer> job = tracker.newJob(KeyValueSource.fromMap(m1));
+            Job<Integer, Integer> job = tracker.newJob(integerKvSource(m1));
             ICompletableFuture<Integer> future = job.mapper(new GroupingTestMapper()).submit(new GroupingTestCollator());
 
             future.andThen(new ExecutionCallback<Integer>() {
@@ -849,7 +849,7 @@ public class MapReduceTest extends HazelcastTestSupport {
             semaphore.acquire();
 
             JobTracker tracker = h1.getJobTracker("default");
-            Job<Integer, Integer> job = tracker.newJob(KeyValueSource.fromMap(m1));
+            Job<Integer, Integer> job = tracker.newJob(integerKvSource(m1));
             ICompletableFuture<Integer> future = job.mapper(new GroupingTestMapper()).reducer(new TestReducerFactory())
                     .submit(new TestCollator());
 
@@ -904,7 +904,7 @@ public class MapReduceTest extends HazelcastTestSupport {
             }
 
             JobTracker jobTracker = h1.getJobTracker("default");
-            Job<Integer, Integer> job = jobTracker.newJob(KeyValueSource.fromMap(m1));
+            Job<Integer, Integer> job = jobTracker.newJob(integerKvSource(m1));
             JobCompletableFuture<Map<String, BigInteger>> future = job.chunkSize(10).mapper(new GroupingTestMapper())
                     .combiner(new ObjectCombinerFactory())
                     .reducer(new ObjectReducerFactory()).submit();
@@ -943,7 +943,7 @@ public class MapReduceTest extends HazelcastTestSupport {
             }
 
             JobTracker jobTracker = h1.getJobTracker("default");
-            Job<Integer, Integer> job = jobTracker.newJob(KeyValueSource.fromMap(m1));
+            Job<Integer, Integer> job = jobTracker.newJob(integerKvSource(m1));
             JobCompletableFuture<Map<String, BigInteger>> future = job.chunkSize(10).mapper(new GroupingTestMapper())
                     .combiner(new ObjectCombinerFactory())
                     .reducer(new NullReducerFactory()).submit();
@@ -973,7 +973,7 @@ public class MapReduceTest extends HazelcastTestSupport {
         }
 
         JobTracker jobTracker = h1.getJobTracker("default");
-        Job<Integer, Integer> job = jobTracker.newJob(KeyValueSource.fromMap(m1));
+        Job<Integer, Integer> job = jobTracker.newJob(integerKvSource(m1));
         ICompletableFuture<Integer> future = job.mapper(new TestMapper())
                 .combiner(new DataSerializableIntermediateCombinerFactory())
                 .reducer(new DataSerializableIntermediateReducerFactory())
@@ -1005,7 +1005,7 @@ public class MapReduceTest extends HazelcastTestSupport {
             }
 
             JobTracker tracker = h1.getJobTracker(randomString());
-            Job<Integer, Employee> job = tracker.newJob(KeyValueSource.fromMap(map));
+            Job<Integer, Employee> job = tracker.newJob(KeyValueSource.<Integer, Employee>fromMap(map));
 
             ICompletableFuture<Map<Integer, Set<Employee>>> future = job
                     .mapper(new ModIdMapper(2))
@@ -1019,6 +1019,10 @@ public class MapReduceTest extends HazelcastTestSupport {
         } finally {
             tripTerminate(h1, h2, h3);
         }
+    }
+
+    static KeyValueSource<Integer, Integer> integerKvSource(IMap<Integer, Integer> m) {
+        return KeyValueSource.<Integer, Integer>fromMap(m);
     }
 
     public static class ModIdMapper implements Mapper<Integer, Employee, Integer, Employee> {

--- a/hazelcast/src/test/java/com/hazelcast/mapreduce/MultiMapMapReduceTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/mapreduce/MultiMapMapReduceTest.java
@@ -77,7 +77,7 @@ public class MultiMapMapReduceTest
         }
 
         JobTracker jobTracker = h1.getJobTracker("default");
-        Job<Integer, Integer> job = jobTracker.newJob(KeyValueSource.fromMultiMap(multiMap));
+        Job<Integer, Integer> job = jobTracker.newJob(KeyValueSource.<Integer, Integer>fromMultiMap(multiMap));
         ICompletableFuture<Map<String, Integer>> ICompletableFuture = job.chunkSize(10).mapper(new MultiMapMapper())
                                                                          .combiner(new MultiMapCombinerFactory())
                                                                          .reducer(new MultiMapReducerFactory()).submit();

